ClosedXML/Excel/XLWorkbook.cs

```diff
@@ -864,21 +864,96 @@ public static Object EvaluateExpr(String expression)
         public XLWorkbook SetLockStructure(Boolean value) { LockStructure = value; return this; }
         public Boolean LockWindows { get; set; }
         public XLWorkbook SetLockWindows(Boolean value) { LockWindows = value; return this; }
-
+        internal HexBinaryValue LockPassword { get; set; }
+        
         public void Protect()
         {
             Protect(true);
         }
 
+        public void Protect(string password)
+        {
+            Protect(true, false, password);
+        }
+
         public void Protect(Boolean lockStructure)
         {
             Protect(lockStructure, false);
         }
 
         public void Protect(Boolean lockStructure, Boolean lockWindows)
         {
+            Protect(lockStructure, lockWindows, null);
+        }
+
+        public void Protect(Boolean lockStructure, Boolean lockWindows, String lockPassword)
+        {
+            if (lockPassword != null)
+            {
+                var hashPassword = GetPasswordHash(lockPassword);
+                if (LockPassword != null)
+                {
+                    if (LockPassword != hashPassword)
+                    {
+                        throw new ArgumentException("Invalid password");
+                    }
+                    else
+                    {
+                        if (lockStructure || lockWindows)
+                        {
+                            throw new InvalidOperationException("The workbook is already protected");
+                        }
+                        else
+                        {
+                            //Unprotect workbook using password.
+                            LockPassword = null;
+                        }
+                    }  
+                }
+                else
+                {
+                    if (lockStructure || lockWindows)
+                    {
+                        //Protect workbook using password.
+                        LockPassword = hashPassword;
+                    }
+                }
+            }
+            else
+            {
+                if (LockPassword != null)

```

Comment:
Lower case variable `lockPassword` is different from the upper case variable `LockPassword`.